### PR TITLE
feat(shorebird_cli): use Google OAuth 2.0

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat: use Google OAuth 2.0 on Shorebird CLI (part of #133)

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/auth/auth.dart [hunk 1]
```diff
@@ -1,54 +1,104 @@
...
+export 'package:googleapis_auth/googleapis_auth.dart' show AccessCredentials;
+
+final _clientId = ClientId(
+  /// Shorebird CLI's OAuth 2.0 identifier.
+  '30552215580-6kgm623s9l4te0rd63r18c5u6b6gkts5.apps.googleusercontent.com',
+
+  /// Shorebird CLI's OAuth 2.0 secret.
+  ///
+  /// This isn't actually meant to be kept secret.
+  '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',
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/auth/auth.dart]: I'm confused.  Is this not supposed to be secret? https://www.oauth.com/oauth2-servers/client-registration/client-id-secret/

Comment (eseidel) [reply packages/shorebird_cli/lib/src/auth/auth.dart]: Looks like the "secret" is optional? https://pub.dev/documentation/googleapis_auth/latest/googleapis_auth/ClientId-class.html

Comment (eseidel) [reply packages/shorebird_cli/lib/src/auth/auth.dart]: https://docs.gitguardian.com/secrets-detection/detectors/specifics/google_oauth2_keys

Comment (felangel) [reply packages/shorebird_cli/lib/src/auth/auth.dart]: > The Client ID must be created with a client_secret here, however there is no way to properly secure a client_secret for installed/console applications. Fortunately the OAuth2 flow used in this case [assumes that the app cannot keep secrets](https://developers.google.com/identity/protocols/oauth2/native-app) so this particular client_secret does not need to be kept secret. You should however make sure not to re-use the same client_secret anywhere secrecy is required.

https://pub.dev/packages/googleapis_auth 

Without a client secret we would get back:

```
Failed to obtain access credentials. Error: invalid_request client_secret is missing. Status code: 400
```

`pub` follows the same approach: https://github.com/dart-lang/pub/blob/196e89e5716d64b2954021e6ac9bb5a20d975fc6/lib/src/oauth2.dart#L37

File: packages/shorebird_cli/lib/src/commands/login_command.dart [hunk 2]
```diff
@@ -18,32 +18,37 @@ class LoginCommand extends ShorebirdCommand {
...
     } catch (error) {
-      loginProgress.fail();
       logger.err(error.toString());
       return ExitCode.software.code;
     }
   }
+
+  void _prompt(String url) {
+    logger.info('''
+Shorebird needs your authorization to manage apps, releases, and patches on your behalf.
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/login_command.dart]: Is "Shorebird" here the command or the service or both?

**Review by eseidel — APPROVED** [review]
Other than the client_secret question lgtm.

File: packages/shorebird_cli/lib/src/commands/login_command.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/login_command.dart]: I was referring to the Shorebird CLI tool in this context. Happy to reword this.

File: packages/shorebird_cli/lib/src/auth/auth.dart (hunk 1 above)
Comment (eseidel) [reply packages/shorebird_cli/lib/src/auth/auth.dart]: OK.  Do we need to leave ourselves more breadcrumbs here about why this isn't secret or what we can/can't do with it?  Thanks.

File: packages/shorebird_cli/lib/src/commands/login_command.dart (hunk 2 above)
Comment (eseidel) [reply packages/shorebird_cli/lib/src/commands/login_command.dart]: I think if it's the `shorebird` command we should say something like:
"The `shorebird` command needs your authorization to manage your Shorebird services on your behalf."

Close enough regardless.  Thanks.

Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/login_command.dart]: Updated ✅

**Pull request merged**